fs.go

```diff
@@ -134,3 +140,47 @@ func requestHdfsData(client http.Client, req http.Request) (HdfsJsonData, error)
 	hdfsData, err := responseToHdfsData(rsp)
 	return hdfsData, err
 }
+
+func (fs *FileSystem) sendHttpRequest(method string, p *Path, params *map[string]string, body io.Reader, byTransport bool) (rsp *http.Response, err error) {
+	var attemptHosts []string
+	defer func() {
+		if rc := recover(); rc != nil {
+			debugStack := ""
+			for _, v := range strings.Split(string(debug.Stack()), "\n") {
+				debugStack += v + "\n"
+			}
+			log.Printf("panic: %v, %v\n", rc, debugStack)
+			return
+		}
+	}()
+start:
```

Comment:
Instead of using a label, this code block could be clearer if the portion that selects the next address is separated into a separate function/method.
Example:
* Loop through your queued address, then attempt connection
* if connection failed, call your function to get next address

如果选择下一个地址的部分被分离到一个单独的函数/方法中，那么这个代码块可能会更清晰，而不是使用标签。
例子：
* 遍历您排队的地址，然后尝试连接
* 如果连接失败，调用你的函数获取下一个地址